test/hotspot/jtreg/compiler/intrinsics/unsafe/ByteBufferTest.java

```diff
@@ -216,7 +216,7 @@ void ck(long x, long y) {
     }
 
     void ck(double x, double y) {
-        if (x != y) {
+        if (x == x && y == y && x != y) {
```

Comment:
This line of logic is begging for a comment of some sort.
Especially to keep someone from coming along later and "optimizing" it.

Reply:
I suggest using `Double::isNan` instead of `x == x`